SQLPractice/src/main/java/com/sql/topic5/ScriptRunner.java

```diff
@@ -0,0 +1,248 @@
+                                                        printlnError(e);
+                                                }
+                                        }
+ 
+                                        if (autoCommit && !conn.getAutoCommit()) {
+                                                conn.commit();
+                                        }
+ 
+                                        ResultSet rs = statement.getResultSet();
+                                        if (hasResults && rs != null) {
+                                                ResultSetMetaData md = rs.getMetaData();
+                                                int cols = md.getColumnCount();
+                                                for (int i = 0; i < cols; i++) {
+                                                        String name = md.getColumnLabel(i);
+                                                        print(name + "\t");
+                                                }
+                                                println("");
+                                                while (rs.next()) {
+                                                        for (int i = 0; i < cols; i++) {
+                                                                String value = rs.getString(i);
+                                                                print(value + "\t");
+                                                        }
+                                                        println("");
+                                                }
+                                        }
+ 
+                                        command = null;
+                                        try {
+                                                statement.close();
+                                        } catch (Exception e) {
+                                                // Ignore to workaround a bug in Jakarta DBCP
+                                        }
+                                        Thread.yield();
+                                } else {
+                                        command.append(line);
+                                        command.append(" ");
+                                }
+                        }
+                        if (!autoCommit) {
+                                conn.commit();
+                        }
+                } catch (SQLException e) {
+                        e.fillInStackTrace();
+                        printlnError("Error executing: " + command);
+                        printlnError(e);
+                        throw e;
+                } catch (IOException e) {
+                        e.fillInStackTrace();
+                        printlnError("Error executing: " + command);
+                        printlnError(e);
+                        throw e;
+                } finally {
+                        conn.rollback();
```

Comment:
High: finally runs always, no matter if the try-catch clause was successful or not. In this case you are rolling back the changes every time.